### PR TITLE
Add PSR3 auto instrumentation

### PR DESCRIPTION
**Pull request by ahkui**
* Add PSR3 auto instrumentation

### CODE REVIEW

File: src/Instrumentation/Psr3/src/Psr3Instrumentation.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            if ($function === 'log') {
+                $level = $params[0] ?? '';
+                $message = $params[1] ?? '';
+                $context = $params[2] ?? [];
+
+                $context['traceId'] = $span->getTraceId();
+                $context['spanId'] = $span->getSpanId();
+
+                return [$level, $message, $context];
+            } else {
```
Comment (pdelewski) [inline src/Instrumentation/Psr3/src/Psr3Instrumentation.php:35]: This else branch is redundant in this case

File: src/Instrumentation/Psr3/src/Psr3Instrumentation.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            if (!$span->isValid()) {
+                return;
+            }
+
+            if ($function === 'log') {
+                $level = $params[0] ?? '';
+                $message = $params[1] ?? '';
+                $context = $params[2] ?? [];
+
+                $context['traceId'] = $span->getTraceId();
```
Comment (pdelewski) [inline src/Instrumentation/Psr3/src/Psr3Instrumentation.php:31]: This code is the same for both branches so can be merged

**Comment by pdelewski** [comment]
@ahkui Could you add some tests?

**Comment by ahkui** [comment]
> @ahkui Could you add some tests?

Ok

File: src/Instrumentation/Psr3/src/Psr3Instrumentation.php (hunk 2 above)
Comment (ahkui) [reply src/Instrumentation/Psr3/src/Psr3Instrumentation.php]: Updated.

**Comment by ahkui** [comment]
> @ahkui Could you add some tests?

UnitTest Added.

**Review by brettmc — CHANGES_REQUESTED** [review]
This looks good. To enable publishing to packagist, can you also add an entry to `.gitsplit.yml` ? I've created https://github.com/opentelemetry-php/contrib-auto-psr3 as the destination.

File: src/Instrumentation/Psr3/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# OpenTelemetry PSR-3 auto-instrumentation
+
```
Comment (brettmc) [inline src/Instrumentation/Psr3/README.md:2]: Could you add a quick overview here of what the instrumentation does? (ie, automatically inject trace id and span id into log message context of any psr3 logger)

**Comment by ahkui** [comment]
> This looks good. To enable publishing to packagist, can you also add an entry to `.gitsplit.yml` ? I've created https://github.com/opentelemetry-php/contrib-auto-psr3 as the destination.

Okay. I will add later

File: src/Instrumentation/Psr3/README.md (hunk 3 above)
Comment (ahkui) [reply src/Instrumentation/Psr3/README.md:2]: updated.

**Comment by ahkui** [comment]
> This looks good. To enable publishing to packagist, can you also add an entry to `.gitsplit.yml` ? I've created https://github.com/opentelemetry-php/contrib-auto-psr3 as the destination.

added.

**Review by brettmc — APPROVED** [review]

**Comment by ahkui** [comment]
thanks.

**Pull request merged**